quickwit-janitor/src/actors/retention_policy_evaluator.rs

```diff
@@ -0,0 +1,460 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+use std::time::Duration;
+
+use async_trait::async_trait;
+use itertools::Itertools;
+use quickwit_actors::{Actor, ActorContext, Handler};
+use quickwit_metastore::{IndexMetadata, Metastore};
+use tracing::{error, info};
+
+use crate::retention_policy_evaluation::run_evaluate_retention_policy;
+
+const RUN_INTERVAL: Duration = Duration::from_secs(60 * 60); // 1 hours
+
+#[derive(Clone, Debug, Default)]
+pub struct RetentionPolicyEvaluatorCounters {
+    /// The number of refresh the config passes.
+    pub num_refresh_passes: usize,
+
+    /// The number of evaluation passes.
+    pub num_evaluation_passes: usize,
+
+    /// The number of deleted splits.
+    pub num_discarded_splits: usize,
+}
+
+#[derive(Debug)]
+struct Loop;
+
+#[derive(Debug)]
+struct Evaluate {
+    index_id: String,
+}
+
+/// An actor for collecting garbage periodically from an index.
+pub struct RetentionPolicyEvaluator {
```

Comment:
I don't think `Evaluator` conveys the fact that the actor also performs a side-effect, i.e. mark some splits for deletion. I'd go with a more explicit name such `RetentionPolicyEnforcer` or `RetentionPolicyExecutor`.

Reply:
I like executor :)

Reply:
Ok then 👍 , curiously I started with Executor then switched to Evaluator.